### PR TITLE
Addresses #4575, API change for Coil:Cooling:DX:SingleSpeed and Coil:Cooling:DX:TwoSpeed

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

 - Addresses #4575 (the last two objects)

checklist:

 - [x] Remove boost::optionals
 - [x] Remove deprecated methods
 - [x] Missing IsAutosized methods
 - [x] Change defaults to requireds
 - [x] Update ctors for requireds

### Pull Request Author

 - [x] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [x] If change to an IDD file, add the label `IDDChange`
 - [x] If breaking existing API, add the label `APIChange`
 - [x] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

**Review by jmarrec — APPROVED** [review]
@joseph-robertson looks good to me.

Could you confirm the regression tests are all still ok?

**Comment by jmarrec** [comment]
@joseph-robertson Also missing an entry in the future Release notes

**Comment by joseph-robertson** [comment]
> @joseph-robertson looks good to me.
> 
> Could you confirm the regression tests are all still ok?

@jmarrec Can you kick them off? (I'm not really sure what the procedure is...)

**Comment by jmarrec** [comment]
> @jmarrec Can you kick them off? (I'm not really sure what the procedure is...)

I was thinking you had the branch built locally and could just run them quickly. 

But there is a workflow dispatch event on the "manual_installer_test" workflow here: https://github.com/NREL/OpenStudio-resources/actions/workflows/manual_installer_test.yml

image

I've triggered a run at https://github.com/NREL/OpenStudio-resources/actions/runs/2907029842

**Comment by jmarrec** [comment]
Not sure what's going on with the 18.04 runner, all tests are segfaulting....

The 20.04 is a conan hiccup, not your doing.

**Comment by jmarrec** [comment]
@joseph-robertson The reg tests are all OS_ASSERTing because this is missing VT rules.

```
[BOOST_ASSERT] <2> Assertion value failed on line 327 of double openstudio::model::detail::CoilCoolingDXSingleSpeed_Impl::latentCapacityTimeConstant() const in file /Users/julien/Software/Others/OpenStudio/src/model/CoilCoolingDXSingleSpeed.cpp.
[BOOST_ASSERT] <2> Assertion value failed on line 284 of double openstudio::model::detail::CoilCoolingDXSingleSpeed_Impl::nominalTimeForCondensateRemovalToBegin() const in file /Users/julien/Software/Others/OpenStudio/src/model/CoilCoolingDXSingleSpeed.cpp.
[BOOST_ASSERT] <2> Assertion value failed on line 299 of double openstudio::model::detail::CoilCoolingDXSingleSpeed_Impl::ratioOfInitialMoistureEvaporationRateAndSteadyStateLatentCapacity() const in file /Users/julien/Software/Others/OpenStudio/src/model/CoilCoolingDXSingleSpeed.cpp.
[BOOST_ASSERT] <2> Assertion value failed on line 313 of double openstudio::model::detail::CoilCoolingDXSingleSpeed_Impl::maximumCyclingRate() in file /Users/julien/Software/Others/OpenStudio/src/model/CoilCoolingDXSingleSpeed.cpp.
[BOOST_ASSERT] <2> Assertion value failed on line 327 of double openstudio::model::detail::CoilCoolingDXSingleSpeed_Impl::latentCapacityTimeConstant() const in file /Users/julien/Software/Others/OpenStudio/src/model/CoilCoolingDXSingleSpeed.cpp.
```

File: src/model/CoilCoolingDXSingleSpeed.cpp [hunk 1]
```diff
@@ -897,18 +923,21 @@ namespace model {
...
     setMaximumOutdoorDryBulbTemperatureForCrankcaseHeaterOperation(0.0);
     setMinimumOutdoorDryBulbTemperatureforCompressorOperation(-25.0);  // Per E+ IDD default
-    //setSupplyWaterStorageTankName("");
-    //setCondensateCollectionWaterStorageTankName("");
+    setNominalTimeForCondensateRemovalToBegin(0.0);
+    setRatioOfInitialMoistureEvaporationRateAndSteadyStateLatentCapacity(0.0);
+    setMaximumCyclingRate(0.0);
+    setLatentCapacityTimeConstant(0.0);
+    setCondenserType("AirCooled");
+    setEvaporativeCondenserEffectiveness(0.9);
```
Comment (jmarrec) [inline src/model/CoilCoolingDXSingleSpeed.cpp:936]: You're changing a default of 0.0 to a default fo 0.9 here...  Yet you're not doing the same in CoilCoolingDXTwoSpeed.

File: src/model/CoilCoolingDXSingleSpeed.cpp [hunk 2]
```diff
@@ -897,18 +923,21 @@ namespace model {
...
+    setCrankcaseHeaterCapacity(0.0);
+    setMaximumOutdoorDryBulbTemperatureForCrankcaseHeaterOperation(10.0);
```
Comment (jmarrec) [inline src/model/CoilCoolingDXSingleSpeed.cpp:938]: setCrankcaseHeaterCapacity is called twice.

setMaximumOutdoorDryBulbTemperatureForCrankcaseHeaterOperation is called twice, one with 0.0 (historical) and later with 10.0. You're also changing a default, and not doing the same on CoilCoolingDXTwoSpeed

File: src/model/CoilCoolingDXSingleSpeed.cpp (hunk 1 above)
Comment (jmarrec) [reply src/model/CoilCoolingDXSingleSpeed.cpp:936]: This is also called twice, oone with 0.0 above (historical) and one with 0.9 now

Comment (jmarrec) [reply src/model/CoilCoolingDXSingleSpeed.cpp:936]: Changing is a default means the regression tests may need to be adjusted to hardsize that field to the historical value or they'll show a deviation...

**Review by jmarrec — COMMENTED** [review]
@joseph-robertson Sorry to bring this back now after merge, but this needs addressing

**Pull request merged**